File: creacionales/factoryMethod/ejemplo.java
interface Transporte { void entregar(); }

class Camion implements Transporte {
    public void entregar() { System.out.println("Entrega por tierra"); }
}

abstract class Logistica {
    abstract Transporte crearTransporte();
    public void planear() {
        Transporte t = crearTransporte();
        t.entregar();
    }
}

class LogisticaTerrestre extends Logistica {
    Transporte crearTransporte() { return new Camion(); }
}

File: creacionales/factoryMethod/ejemplo.py
from abc import ABC, abstractmethod

class Transporte(ABC):
    @abstractmethod
    def entregar(self): pass

class Camion(Transporte):
    def entregar(self): print("Entrega por tierra")

class Logistica(ABC):
    @abstractmethod
    def crear_transporte(self): pass

    def planear(self):
        t = self.crear_transporte()
        t.entregar()

class LogisticaTerrestre(Logistica):
    def crear_transporte(self): return Camion()